packages/flutter_tools/lib/src/windows/visual_studio.dart

```diff
@@ -2,16 +2,33 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-import '../base/context.dart';
+import 'package:meta/meta.dart';
+import 'package:platform/platform.dart';
+import 'package:process/process.dart';
+
+import '../base/file_system.dart';
 import '../base/io.dart';
+import '../base/logger.dart';
 import '../base/process.dart';
 import '../convert.dart';
-import '../globals.dart' as globals;
-
-VisualStudio get visualStudio => context.get<VisualStudio>();
 
 /// Encapsulates information about the installed copy of Visual Studio, if any.
 class VisualStudio {
+  VisualStudio({
+    @required this.fileSystem,
+    @required this.processManager,
+    @required Platform platform,
+    @required Logger logger,
+  }) : _platform = platform,
+       _processUtils = ProcessUtils(processManager: processManager, logger: logger);
+
+  @visibleForTesting
```

Comment:
This seems like a red flag, since it means that the tests are essentially using the object under test as a fixture. I think in the short term, keeping the FileSystem and ProcessManager in a test fixture would be better than this approach, since it limits the remaining debt to the test file. (See other file comments for longer term suggestions.)